### PR TITLE
fix(settings): re-add border to integrations list

### PR DESCRIPTION
**Pull request by Chrissy**
image

image

I'm not in love with this solution, but I also didn't feel like refactoring a ton of tests. I tried tightening up the html to see if I could get by, but a bunch of tests did break. I spiked for maybe 30 minutes trying to fix the tests, before deciding it wasn't worth it.

### CODE REVIEW

File: src/sentry/static/sentry/app/components/panels/panelItem.jsx [hunk 1]
```diff
@@ -1,18 +1,21 @@
...
 import styled from 'react-emotion';
+import isPropValid from '@emotion/is-prop-valid';
 
-const PanelItem = styled(Flex)`
+const PanelItem = styled(Flex, {shouldForwardProp: isPropValid})`
   border-bottom: 1px solid ${p => p.theme.borderLight};
 
   &:last-child {
-    border: 0;
+    border: ${p => (p.forceBorder ? null : 0)};
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/panels/panelItem.jsx]: Should the entire `border` property be conditional?

Comment (Chrissy) [reply src/sentry/static/sentry/app/components/panels/panelItem.jsx]: unfortunately the linter chews it up. 

The compiler omits invalid css lines, so returning `null` will cause this to only show up for classes where `forceBorder` is true.

File: src/sentry/static/sentry/app/views/organizationIntegrations/providerRow.jsx [hunk 2]
```diff
@@ -97,7 +97,12 @@ export default class ProviderRow extends React.Component {
 
   render() {
     return (
-      <PanelItem p={0} direction="column" data-test-id={this.props.provider.key}>
+      <PanelItem
+        p={0}
+        forceBorder={true}
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/views/organizationIntegrations/providerRow.jsx]: You can shorthand this as just `forceBorder` without the `={true}`

**Comment by EvanPurkhiser** [comment]
There's an extra border at the bottom.

File: src/sentry/static/sentry/app/components/panels/panelItem.jsx [hunk 3]
```diff
@@ -1,18 +1,25 @@
...
+  border-bottom:  1px solid ${p.theme.borderLight};
 
   &:last-child {
-    border: 0;
+    border-bottom: 0;
   }
 `;
 
+const PanelItem = styled(Flex, {shouldForwardProp: isPropValid})`
+  ${p => !p.borderless && getBorders(p)};
```
Comment (Chrissy) [inline src/sentry/static/sentry/app/components/panels/panelItem.jsx]: solid syntax highlighting here

Comment (billyvg) [reply src/sentry/static/sentry/app/components/panels/panelItem.jsx]: you could do
```suggestion
  ${getBorders};
```

And handle it in `getBorders`

**Comment by Chrissy** [comment]
@EvanPurkhiser Because we don't really know the order of these items, it is hard to remove the border from the last element in a way that isn't easy to break.

I am intent to not spend my entire day on this, so I rewrote the pull using a negative margin hack that basically collapses the top border into the header. The screenshot is updated and shows what it looks like now.

**Comment by Chrissy** [comment]
closing in favor of https://github.com/getsentry/sentry/pull/13210

**Comment by Chrissy** [comment]
closing in favor of https://github.com/getsentry/sentry/pull/13210

**Pull request closed without merging**